src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -35,6 +37,48 @@ public AnnotationConfigApplicationContext(String packageName) {
     private void scan() {
     }
 
+    private void inject() {
+        // inject by fields
+        context.values().forEach(this::injectBeanWithFields);
+        // TODO: inject by setters
+    }
+
+    /**
+     * Inject bean fields with @Autowired annotation.
+     * Called after filling the context and instances initializing
+     *
+     * @param bean Bean for injection
+     * @param <T>
+     */
+    private <T> void injectBeanWithFields(T bean) {
+        Field[] fields = bean.getClass().getDeclaredFields();
+        Arrays.stream(fields)
+                .filter(field -> !field.getType().isPrimitive() && field.isAnnotationPresent(Autowired.class))
```

Comment:
Should we filter if a class has a @Component annotation?

Reply:
No. We can create an objects in our package outside of the context but with using fields from this context.

Reply:
Ok. I see that there is no need to filter @Component classes because you work with classes that are already were filtered.